src/ngraph/node.hpp

```diff
@@ -37,30 +37,30 @@
 #include "ngraph/node_vector.hpp"
 #include "ngraph/placement.hpp"
 
+// There is work in progress on deprecating some old APIs here, and for development purposes it is
+// nice to have a compile-time option to turn on actual deprecation.
+//
+// Enable the deprecation with -DNGRAPH_DEPRECATE_OLD_NODE_APIS=ON at cmake time.
+#ifdef NGRAPH_DEPRECATE_OLD_NODE_APIS
```

Comment:
This could be made more general by calling it NGRAPH_DEPRECATED and put it in deprecated.hpp. Then it could be used in other places where it really should be used.

Reply:
I like it.

Reply:
Have created an `NGRAPH_DEPRECATED` macro in `deprecated.hpp`, enabled at cmake time with `-DNGRAPH_DEPRECATED_ENABLE=ON`.